shell/common/shell.cc

```diff
@@ -542,6 +542,14 @@ bool Shell::Setup(std::unique_ptr<PlatformView> platform_view,
   weak_rasterizer_ = rasterizer_->GetWeakPtr();
   weak_platform_view_ = platform_view_->GetWeakPtr();
 
+  // Setup the time-consuming default font manager right after engine created.
+  fml::TaskRunner::RunNowOrPostTask(task_runners_.GetUITaskRunner(),
+                                    [engine = weak_engine_] {
+                                      if (engine) {
+                                        engine->SetupDefaultFontManager();
```

Comment:
What protection do we have that someone won't use `FontCollection.collection_` before this executes?

Reply:
The `FontCollection` is only usable on the UI thread, and UI thread tasks are executed in the order that they were queued.  Therefore the task that completes setup of the `FontCollection` will execute before any task that might query the `FontCollection`.

Reply:
There could already be something in the UI event queue that will use the FontCollection though.
```
platform_thread->execute({
  ui_thread->execute({ shell->setup(); });
  ui_thread->execute({ someFontCollectionUsingFunction(); });
});
```

There is no guarantee in this case that `SetupDefaultFontManager` will execute before `someFontCollectionUsingFunction`.

Reply:
The embedder APIs will not return a shell to the caller until `Shell::Setup` has completed and the `SetupDefaultFontManager` task has already been queued.